packages/repository/src/__tests__/integration/repositories/relation.factory.integration.ts

```diff
@@ -413,6 +413,32 @@ describe('HasManyThrough relation', () => {
       ]),
     );
   });
+
+  it('links a target instance to the source instance', async () => {
+    const item = await cartItemRepo.create({description: 'an item'});
+    let targets = await customerCartItemRepo.find();
+    expect(targets).to.deepEqual([]);
+
+    await customerCartItemRepo.link(item, {throughData: {id: 98}});
```

Comment:
I am confused about the particular `throughData` value used here. Is it necessary to set the primary key `id` in order to make `link` work? IMO, `id` should be almost always generated by the database. If your intention is to verify that `throughData` parameter is honored, then please set a different property - one that's neither primary nor foreign key.

Reply:
Oh I thought the id couldn't be auto-generated. Will remove it from all the integration tests.